packages/adblocker-benchmarks/run.js

```diff
@@ -89,10 +89,18 @@ function loadLists() {
   // We still use old versions of EasyList and EasyPrivacy whereas current
   // versions no longer use $object-subrequest
   // https://gitlab.com/eyeo/adblockplus/abc/adblockpluscore/-/issues/6
-  content = content.replace(/(?!$.*)\bobject-subrequest\b/g, 'object');
+  content = content.replace(/(?!$.*)\bobject-subrequest\b/gm, 'object');
 
+  // Cliqz considers a filter like /foo^$domain=|example.com to be invalid.
   // https://github.com/cliqz-oss/adblocker/discussions/2114#discussioncomment-1133958
-  content = content.replace(/(?!$.*)\bdomain=\|/g, 'domain=');
+  content = content.replace(/(?!$.*)\bdomain=\|/gm, 'domain=');
+
+  // Remove filters with regular expression patterns containing lookahead and
+  // lookbehind assertions.
+  // https://github.com/cliqz-oss/adblocker/discussions/2114#discussioncomment-1135161
+  //
+  // Note: The regular expression below is not right, but it does the job.
+  content = content.replace(/^(@@)?\/.*\(\?.*/gm, '');
```

Comment:
How many such filters are there in the lists?

Reply:
There are 3 such filters:

```
/^((?!(^https?):\/\/(ajax\.googleapis\.com|cdnjs\.cloudflare\.com|fonts\.googleapis\.com)\/).*)$/$script,third-party,domain=oladblock.me|oladblock.services|oladblock.xyz|oload.cc|oload.cloud|oload.club|oload.download|oload.fun|oload.info|oload.live|oload.press|oload.services|oload.site|oload.space|oload.stream|oload.tv|oload.win|olpair.com|openload.cc|openload.co|openload.fun|openload.pw|openloed.co
/^https?:\/\/([0-9a-z\-]+\.)?(9anime|animeland|animenova|animeplus|animetoon|animewow|gamestorrent|goodanime|gogoanime|igg-games|kimcartoon|memecenter|readcomiconline|toonget|toonova|watchcartoononline)\.[a-z]{2,4}\/(?!([Ee]xternal|[Ii]mages|[Ss]cripts|[Uu]ploads|ac|ajax|assets|combined|content|cov|cover|(img\/bg)|(img\/icon)|inc|jwplayer|player|playlist-cat-rss|static|thumbs|wp-content|wp-includes)\/)(.*)/$image,other,script,~third-party,xmlhttprequest,domain=~animeland.hu|~memecenter.fr
/http://[a-zA-Z0-9]+\.[a-z]+\/.*(?:[!"#$%&'()*+,:;<=>?@/\^_`{|}~-]).*[a-zA-Z0-9]+/$script,third-party,domain=keezmovies.com|redtube.com|tube8.com|tube8.es|tube8.fr|www.pornhub.com|youporn.com
```

Should we remove them from the list itself?

Then the same question arises for the other two cases. We could edit the list itself to fix those filters. But in the case of `$object-subrequest` there are 462 such filters. In the `$domain=|` case there is only one.

Reply:
Rewriting looks ok. I was just curious about this being a corner case or widely seen pattern.